wordpress-mobile/react-native-recyclerview-list
# Feature/match wpandroid api level

This PR:
* Upgrades the gradle plugin and gradle wrapper
* Bumps the `compileSdkVersion`, `buildToolsVersion` and `targetSdkVersion` to match current wpandroid's ones, in preparation of integrating the mobile Gutenberg editor with wpandroid
* Bumps the support library version to `27.1.1`
* Adapts the implementation of `NotAnimatedItemAnimator` to the upgraded support library version

### To test (A):
* Run the example app and it should work fine.

### To test (B):
* Use the https://github.com/wordpress-mobile/gutenberg-mobile/pull/201 PR to try out the lib inside the mobile Gutenberg app. It should also work as usual there.
